src/test/ui/parser/struct-literal-in-match-discriminant.rs

```diff
@@ -1,15 +1,13 @@
-// compile-flags: -Z parse-only
-
 struct Foo {
     x: isize,
 }
 
 fn main() {
-    match Foo {
+    match Foo { //~ ERROR expected value, found struct `Foo`
         x: 3    //~ ERROR expected one of `=>`, `@`, `if`, or `|`, found `:`
     } {
```

Comment:
This ideally would be recovered to avoid the knock down errors, but that's unlikely to happen soon.